### PR TITLE
br: fix insert gc failed due to slow schema reload

### LINKED ISSUE #57743: br: wait till info schema loading finished before proceeding to the rest of the PiTR restore
## Bug Report

In a cluster with large number of tables but minimal actual data, during PiTR restore infoSchema can take minutes to load the restore change, and since actual data restore finished within seconds, restore process tries to use infoSchema for insertGCRow before its reloading finished, and it can cause problem of the below 
```
[error="[domain:8027]Information schema is out of date: schema failed to update in 1 lease, please make sure TiDB can connect to TiKV"]
```

### PR DESCRIPTION
**Pull request by Tristan1900**
### What problem does this PR solve?

Issue Number: close #57743

Problem Summary:

### What changed and how does it work?
Wait until schema info finished reloading before proceeding to the rest of the restore process,
Added test to first to reproduce the issue and then add the fix to verify the issue has been fixed (TDD)
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fixed loading schema taking long time during restore causing insertGc row and other operations to fail  
```

### CODE REVIEW

File: pkg/domain/domain.go [hunk 1]
```diff
@@ -1513,6 +1522,12 @@ func (do *Domain) GetSchemaLease() time.Duration {
 	return do.schemaLease
 }
 
+// IsLeaseExpired returns whether lease has expired
+func (do *Domain) IsLeaseExpired() bool {
+	startReloadTime := time.Now()
+	return time.Since(startReloadTime) > do.schemaLease
```
Comment (3pointer) [inline pkg/domain/domain.go:1528]: this function seems never return true?
because time.Since(time.Now()) is close to zero.

Comment (Tristan1900) [reply pkg/domain/domain.go:1528]: you are absolutely right, that's a dumb mistake. but why did the test passed, let me take a look!

File: br/pkg/task/stream.go [hunk 2]
```diff
@@ -1493,6 +1499,21 @@ func restoreStream(
...
+	failpoint.Inject("post-restore-kv-pending", func(val failpoint.Value) {
+		if val.(bool) {
+			// not ideal to use sleep but not sure what's the better way right now
+			log.Info("sleep after restoring kv")
+			time.Sleep(2 * time.Second)
+		}
+	})
+
+	// make sure schema reload finishes before proceeding
+	if err = waitUntilSchemaReload(ctx, client); err != nil {
```
Comment (3pointer) [inline br/pkg/task/stream.go:1513]: how about put `waitUntilSchemaReload` after `CleanUpKVFiles`?

**Review by 3pointer — APPROVED** [review]
rest LGTM

**Review by Leavrth — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]
domain part lgtm

**Pull request merged**